### PR TITLE
middleware2d components rendering in native mode

### PR DESCRIPTION
**Pull request by zhakesi**

### CODE REVIEW

File: native/cocos/editor-support/MeshBuffer.cpp [hunk 1]
```diff
@@ -77,6 +93,9 @@ void MeshBuffer::uploadVB() {
     auto *rVB = _vbArr[_bufferPos];
     rVB->reset();
     rVB->writeBytes(reinterpret_cast<const char *>(_vb.getBuffer()), _vb.length());
+    UIMeshBuffer *uiMeshBuffer = (UIMeshBuffer *)_uiMeshBufferArr[_bufferPos];
+    uiMeshBuffer->setVData((float*)rVB->getBuffer());
```
Comment (minggo) [inline native/cocos/editor-support/MeshBuffer.cpp]: Please use static_cast

File: native/cocos/editor-support/MeshBuffer.cpp [hunk 2]
```diff
@@ -107,4 +132,31 @@ void MeshBuffer::reset() {
...
+        attrs = ATTRIBUTES_V3F_T2F_C4B;
+    } else {
+        attrs = ATTRIBUTES_V3F_T2F_C4B_C4B;
+    }
+    uiMeshBuffer->initialize(std::move(attrs), true);
+    _uiMeshBufferArr.push_back(uiMeshBuffer);
+}
+
+void MeshBuffer::cleanUIMeshBuffer() {
+    for(auto &buf : _uiMeshBufferArr) {
```
Comment (minggo) [inline native/cocos/editor-support/MeshBuffer.cpp]: auto &buf -> auto *buf

File: native/cocos/editor-support/MeshBuffer.h [hunk 3]
```diff
@@ -84,6 +88,10 @@ class MeshBuffer {
         return _ib;
     }
 
+    cc::UIMeshBuffer *getUIMeshBuffer();
```
Comment (minggo) [inline native/cocos/editor-support/MeshBuffer.h]: cc::UIMeshBuffer *getUIMeshBuffer() -> cc::UIMeshBuffer *getUIMeshBuffer() const

File: native/cocos/editor-support/MeshBuffer.h [hunk 4]
```diff
@@ -84,6 +88,10 @@ class MeshBuffer {
...
+
+    const ccstd::vector<cc::UIMeshBuffer *> &uiMeshBuffers();
```
Comment (minggo) [inline native/cocos/editor-support/MeshBuffer.h]: const ccstd::vector<cc::UIMeshBuffer *> &uiMeshBuffers() -> const ccstd::vector<cc::UIMeshBuffer *> &uiMeshBuffers() const

File: native/cocos/editor-support/MiddlewareManager.cpp [hunk 5]
```diff
@@ -141,6 +120,18 @@ void MiddlewareManager::render(float dt) {
...
             buffer->uploadVB();
         }
+
+        uint16_t accID = 65534;
+        auto* batch2d = Root::getInstance()->getBatcher2D();
+        if (it.first == VF_XYZUVCC) {
+            accID = 65535;
+        }
+        ccstd::vector<UIMeshBuffer *> uiMeshArray;
+        auto uiBufArray = buffer->uiMeshBuffers();
```
Comment (minggo) [inline native/cocos/editor-support/MiddlewareManager.cpp]: Use reference instead

File: native/cocos/editor-support/dragonbones-creator-support/CCArmatureDisplay.cpp [hunk 6]
```diff
@@ -438,19 +396,54 @@ se_object_ptr CCArmatureDisplay::getSharedBufferOffset() const {
     return nullptr;
 }
 
-se_object_ptr CCArmatureDisplay::getParamsBuffer() const {
-    if (_paramsBuffer) {
-        return _paramsBuffer->getTypeArray();
+uint32_t CCArmatureDisplay::getRenderOrder() const {
+    return 0;
```
Comment (minggo) [inline native/cocos/editor-support/dragonbones-creator-support/CCArmatureDisplay.cpp]: Always return 0?

Comment (zhakesi) [reply native/cocos/editor-support/dragonbones-creator-support/CCArmatureDisplay.cpp]: no used.

Comment (minggo) [reply native/cocos/editor-support/dragonbones-creator-support/CCArmatureDisplay.cpp]: Can delete it?

Comment (zhakesi) [reply native/cocos/editor-support/dragonbones-creator-support/CCArmatureDisplay.cpp]: done

**Review by minggo — APPROVED** [review]

**Review by SantyWang — APPROVED** [review]

**Review by SantyWang — APPROVED** [review]

**Review by Zhmz — APPROVED** [review]

File: native/cocos/2d/renderer/UIMeshBuffer.cpp [hunk 7]
```diff
@@ -172,6 +172,11 @@ void UIMeshBuffer::parseLayout() {
     _meshBufferLayout = reinterpret_cast<MeshBufferLayout*>(_sharedBuffer);
 }
 
+// only initialize needCreateLayout = true call it
+void UIMeshBuffer::freeLayout() {
```
Comment (minggo) [inline native/cocos/2d/renderer/UIMeshBuffer.cpp]: Should be public? It seems is used internally.

Comment (minggo) [reply native/cocos/2d/renderer/UIMeshBuffer.cpp]: And can free it in destructor?

Comment (zhakesi) [reply native/cocos/2d/renderer/UIMeshBuffer.cpp]: done

**Review by minggo — APPROVED** [review]

**Review by SantyWang — APPROVED** [review]

**Pull request merged**